### DISCUSSION TITLE
In an op, is there a way to handle a cancellation event?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17686] [Q&A] [answered]
Hi, is there a way to detect in an op if a dagster cancellation event was sent? I want to run some code before letting the op get terminated

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699011091135069?thread_ts=1699011091.135069&cid=C01U954MEER).*

**Comment by gibsondan** [comment, marked as answer]
This will result in a `DagsterExecutionInterruptedError` being raised in your code - this exception can be caught and handled if you want to do cleanup (you should be sure to re-raise it though so that the job can still terminate).

Note that `DagsterExecutionInterruptedError` subclasses `BaseException` in Python but does not subclass `Exception` (much like `KeyboardInterrupt` in Python which is raised when you CTRL-C a program in the terminal).

For example:
```
@asset
def my_databricks_asset():
  try:
    run_id = submit_databricks_job(...)
  except DagsterExecutionInterruptedError as e:
    cleanup()
    raise e
```
